### PR TITLE
✨ Add Custom Net Device

### PR DESCRIPTION
**Pull request by YousefEZ**
This PR is aimed to address the issue with packets dropping when attempting to reroute. Also its marks the separation of the FRR Policy utilization by the Queue and instead it moves it up to the net device.

### CODE REVIEW

File: libs/point_to_point_frr_helper.h [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        return;
+    }
+
+    PcapHelper pcapHelper;
+
+    std::string filename;
+    if (explicitFilename) {
+        filename = prefix;
+    } else {
+        filename = pcapHelper.GetFilenameFromDevice(prefix, device);
```
Comment (charliebarber) [inline libs/point_to_point_frr_helper.h:143]: Does this mean we can mess with the file names a bit? Might help to clear up the different suffixes to know which gateway we are looking at.

File: libs/point_to_point_frr_helper.h
Comment (charliebarber) [inline libs/point_to_point_frr_helper.h:1]: This one is a little bit beyond me but I assume a lot is taken from NS3 source.

File: libs/point_to_point_frr_net_device.h [hunk 2]
```diff
@@ -0,0 +1,1161 @@
...
+    }
+    return 0;
+}
+
+template <typename FRR_POLICY>
+uint16_t PointToPointFRRNetDevice<FRR_POLICY>::EtherToPpp(uint16_t proto)
+{
+    NS_LOG_FUNCTION_NOARGS();
+    switch (proto) {
+    case 0x0800: return 0x0021; // IPv4
```
Comment (charliebarber) [inline libs/point_to_point_frr_net_device.h:1108]: Seeing the protocol numbers that clearly in front of me is slightly infuriating.....

**Review by charliebarber — APPROVED** [review]
LGTM. Somewhat confused by the p2p code but I guess I do not need to understand all of it. Well done as always.

File: libs/point_to_point_frr_net_device.h (hunk 2 above)
Comment (YousefEZ) [reply libs/point_to_point_frr_net_device.h:1108]: yeah I was wondering where I got the number from 😂😂

File: libs/point_to_point_frr_helper.h
Comment (YousefEZ) [reply libs/point_to_point_frr_helper.h:1]: most of it yeah, just modifying to create PointToPointFRRNetDevices instead

**Pull request merged**